src/Core/Transform/MutualInformationAffine/Code.py

```diff
@@ -0,0 +1,102 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# =========================================================================*/
+
+import sys
+import numpy as np
+
+import itk
+
+if len(sys.argv) < 4:
+    raise Exception(f"Usage: {sys.argv[0]} fixed_image moving_image output_image")
+
+# Import images
+fixed_image = itk.imread(sys.argv[1], itk.UC)
+moving_image = itk.imread(sys.argv[2], itk.UC)
+
+# Preprocess images
+fixed_normalized_image = itk.normalize_image_filter(fixed_image)
+fixed_smoothed_image = itk.discrete_gaussian_image_filter(
+    fixed_normalized_image, variance=2.0
+)
+
+moving_normalized_image = itk.normalize_image_filter(moving_image)
+moving_smoothed_image = itk.discrete_gaussian_image_filter(
+    moving_normalized_image, variance=2.0
+)
+
+# Run gradient ascent
+transform = itk.AffineTransform[itk.D, fixed_image.GetImageDimension()].New()
+interpolator = itk.LinearInterpolateImageFunction[type(fixed_image), itk.D].New()
+
+metric = itk.MutualInformationImageToImageMetric[
+    type(fixed_image), type(moving_image)
+].New()
+
+metric.SetNumberOfSpatialSamples(100)
+metric.SetFixedImageStandardDeviation(5.0)
+metric.SetMovingImageStandardDeviation(5.0)
+
+metric.ReinitializeSeed(121212)
+
+optimizer = itk.GradientDescentOptimizer.New()
+
+optimizer.SetLearningRate(1.0)
+optimizer.SetNumberOfIterations(200)
+optimizer.MaximizeOff()
+
+# Set scales so that the optimizer can take
+# large steps along translation parameters,
+# moderate steps along rotational parameters, and
+# small steps along scale parameters
+optimizer.SetScales([100, 0.5, 0.5, 100, 0.001, 0.001])
+
+registrar = itk.ImageRegistrationMethod[type(fixed_image), type(moving_image)].New()
+registrar.SetFixedImage(fixed_smoothed_image)
+registrar.SetMovingImage(moving_smoothed_image)
+registrar.SetTransform(transform)
+registrar.SetInterpolator(interpolator)
+registrar.SetMetric(metric)
+registrar.SetOptimizer(optimizer)
+
+registrar.SetFixedImageRegion(fixed_image.GetBufferedRegion())
+registrar.SetInitialTransformParameters(transform.GetParameters())
+
+registrar.Update()
+
+# Print final results
+print(f"Its: {optimizer.GetCurrentIteration()}")
+print(f"Final Value: {optimizer.GetValue()}")
+print(f"Final Position: {list(registrar.GetLastTransformParameters())}")
+
+# Resample and write out image
+ResampleFilterType = itk.ResampleImageFilter[type(fixed_image), type(fixed_image)]
+resample = ResampleFilterType.New(
+    Transform=transform,
+    Input=moving_image,
+    Size=fixed_image.GetLargestPossibleRegion().GetSize(),
+    OutputOrigin=fixed_image.GetOrigin(),
+    OutputSpacing=fixed_image.GetSpacing(),
+    OutputDirection=fixed_image.GetDirection(),
+    DefaultPixelValue=100,
+)
+
+resample.Update()
```

Comment:
Let's use the simpler, functional interface:

```
resampled = itk.resample_image_filter(moving_image, output_origin=fixed_image.GetOrigin(), [...]
```

Reply:
Same type issue as noted above. Functional interface seems to expect images of type `itk.SS`.
> Traceback (most recent call last):
  File "Code.py", line 93, in <module>
    resampled = itk.resample_image_filter(
  File "C:\repos\ITK-builds-2\Wrapping\Generators\Python\itk\support\itkHelpers.py", line 211, in image_filter_wrapper
    return image_filter(*args, **kwargs)
  File "C:\repos\ITK-builds-2\Wrapping\Generators\Python\itk\itkResampleImageFilterPython.py", line 1927, in resample_image_filter
    instance = itk.ResampleImageFilter.New(*args, **kwargs)
  File "C:\repos\ITK-builds-2\Wrapping\Generators\Python\itk\support\itkTemplate.py", line 726, in New
    return self[list(keys)[0]].New(*args, **kwargs)
  File "C:\repos\ITK-builds-2\Wrapping\Generators\Python\itk\itkResampleImageFilterPython.py", line 680, in New
    itkTemplate.New(obj, *args, **kargs)
  File "C:\repos\ITK-builds-2\Wrapping\Generators\Python\itk\support\itkTemplate.py", line 792, in New
    itk.set_inputs(self, args, kargs)
  File "C:\repos\ITK-builds-2\Wrapping\Generators\Python\itk\support\itkExtras.py", line 928, in set_inputs
    attrib(itk.output(value))
TypeError: Expecting argument of type itkImageSS2 or itkImageSourceISS2.
Additional information:
Wrong number or type of arguments for overloaded function 'itkImageToImageFilterISS2ISS2_SetInput'.
  Possible C/C++ prototypes are:
    itkImageToImageFilterISS2ISS2::SetInput(itkImageSS2 const *)
    itkImageToImageFilterISS2ISS2::SetInput(unsigned int,itkImageSS2 const *)

Reply:
What is `type(moving_image)`?

Reply:
> \> print(type(moving_image))
<class 'itk.itkImagePython.itkImageUC2'>

Note that no error is thrown when reading in images of type itk.SS but `itk.imwrite` does not support writing `itk.SS` PNG output images by default.

Per above, this may be resolved by updating to the v4 metric.